plugin/evm/gossiper_eth_gossiping_test.go

```diff
@@ -113,26 +117,9 @@ func TestMempoolEthTxsAppGossipHandling(t *testing.T) {
 	// Txs must be submitted over the API to be included in push gossip.
 	// (i.e., txs received via p2p are not included in push gossip)
 	require.NoError(vm.eth.APIBackend.SendTx(context.Background(), tx))
+	require.NoError(vm.ethTxPushGossiper.Get().Gossip(context.Background()))
+	statusLock.Lock()
```

Comment:
This lock _is_ actually doing something, but it's only ensuring that there are no calls to either `tvm.AppSender.SendAppRequestF` or `tvm.AppSender.SendAppGossipF` between the two checks here, which I'm not sure if that's the desired effect.